aten/src/ATen/native/sparse/SparseBlasImpl.cpp

```diff
@@ -70,6 +75,31 @@ Tensor& _compressed_row_strided_mm_out(const Tensor& compressed, const Tensor& s
     blocksize = {values.size(-2), values.size(-1)};
   }
 
+// No stable support for ROCM in Triton yet.
+#ifndef USE_ROCM
+  // Triton works only with blocksizes which are powers of 2.
+  const auto is_power_of_2 = [](int64_t v) -> bool {
+    return !(v & (v - 1));
+  };
+
+  // Dtype and blocksize checks for potential Triton usage.
+  if ((strided.scalar_type() == ScalarType::Half
+    || strided.scalar_type() == ScalarType::BFloat16)
+   && is_power_of_2(blocksize[0]) && is_power_of_2(blocksize[1])
+   && (blocksize[0] >= 16) && (blocksize[1] >= 16)
+   // lhs is retiled to (b0, b1) while rhs is to (b1, b0),
+   // so the result is tiled to (b0, b0) and we need to make
+   // sure that dense.size(-1) is divisible by b0.
+   && n % blocksize[0] == 0) {
+    const auto triton_kernel = c10::Dispatcher::singleton()
+      .findOp(torch::jit::parseName("aten::_triton_bsr_dense_mm"));
```

Comment:
Is there a way to avoid creating a dependency on torch/csrc here?

cc @albanD @nikitaved

Reply:
Idea: Rather than calling out to the triton kernel from here when appropriate, the call point we register in python land could call into this if the triton kernel is not appropriate.

Basically the wrapper around the triton kernel in python would be used to replace dispatch for `addmm` with the appropriate dispatch key. That is assuming it could stash the existing dispatch target as a callable somewhere. 

That would avoid the dependency referenced by @cpuhrsch, but I am not sure if it would work.

Reply:
You don't need jit code here: `c10::Dispatcher::singleton().findSchemaOrThrow("aten::_triton_bsr_dense_mm", "")` should work here no?

Reply:
^ That is much more simple, I was operating under the assumption that the JIT was required here.

Reply:
Neither of these things I liked to begin with, it is just easier to create a native function that returns nothing or something. And that will spare us from the schema search.

Reply:
@albanD , how do I check the dispatch key with this?